src/Moq/Setup.cs

```diff
@@ -65,7 +65,25 @@ public void Execute(Invocation invocation)
 			this.Condition?.SetupEvaluatedSuccessfully();
 			this.expectation.SetupEvaluatedSuccessfully(invocation);
 
-			this.ExecuteCore(invocation);
+			if (this.expectation.HasResultExpression(out var awaitableFactory))
+			{
+				try
+				{
+					this.ExecuteCore(invocation);
+				}
+				catch (Exception exception)
+				{
+					invocation.Exception = exception;
+				}
```

Comment:
We may have to change this some day in order to account for several / aggregate exceptions.